ports/mimxrt10xx/common-hal/busio/I2C.c

```diff
@@ -177,6 +164,11 @@ void common_hal_busio_i2c_deinit(busio_i2c_obj_t *self) {
 
     self->sda = NULL;
     self->scl = NULL;
+    common_hal_busio_i2c_mark_deinit(self);
```

Comment:
redundant assignment also done in mark_deinit

Reply:
Fixed all of these, and also added
```c
    // Ensure the object starts in its deinit state.
    common_hal_busio_i2c_mark_deinit(self);
```
at the top of each constructor. Strictly speaking, this is only necessary for `NO_PIN` deinit markers, which are -1, not 0, but for safety and consistency I did them all.